packages/analytics-core/src/plugins/destination.ts

```diff
@@ -297,11 +297,24 @@ export class Destination implements DestinationPlugin {
    * 1) new events are added to queue; or
    * 2) response comes back for a request
    */
-  saveEvents() {
+  async saveEvents() {
     if (!this.config.storageProvider) {
       return;
     }
+
     const events = Array.from(this.queue.map((context) => context.event));
-    void this.config.storageProvider.set(this.storageKey, events);
+    const storedEvents = (await this.config.storageProvider.get(this.storageKey)) || [];
+
+    let deletedEvents = [];
+    if (storedEvents && storedEvents.length + events.length > this.config.savedMaxCount) {
+      const removeCount = storedEvents.length + events.length - this.config.savedMaxCount;
+      deletedEvents = storedEvents.splice(0, removeCount);
+      this.config.loggerProvider.error(
+        `Exceeded maximum saved events count. ${deletedEvents.length} events are dropped.`,
+      );
+    }
+
+    const combinedEvents = [...events, ...storedEvents];
+    await this.config.storageProvider.set(this.storageKey, combinedEvents);
```

Comment:
all the events we plan to persist isare in `events` array already and no need to read from storage. can we just truncate the `events` array to enforce the limit?

```suggestion
void this.config.storageProvider.set(this.storageKey, events.slice(0, this.savedMaxCount);
```

Reply:
Thank you Kevin. It's really helpful to know that.